src/GafferCortex/CompoundParameterHandler.cpp

```diff
@@ -105,33 +105,27 @@ Gaffer::Plug *CompoundParameterHandler::setupPlug( Gaffer::GraphComponent *plugP
 
 	setupPlugFlags( m_plug.get(), flags );
 
-	// remove any child plugs we don't need
-
+	// loop through the handlers and remove any that are not linked to a new parameter
+	const CompoundParameter::ParameterVector &children = m_parameter->orderedParameters();
 	std::vector<Gaffer::PlugPtr> toRemove;
-	for( Gaffer::PlugIterator pIt( m_plug.get() ); !pIt.done(); ++pIt )
+	for( HandlerMap::iterator it = m_handlers.begin(), eIt = m_handlers.end(); it!=eIt; )
 	{
-		if( (*pIt)->getName().string().compare( 0, 2, "__" ) == 0 )
-		{
-			// we leave any plugs prefixed with __ alone, on the assumption
-			// that they don't represent child parameters but instead are
-			// used for bookkeeping by a derived parameter handler (ClassParameterHandler
-			// or ClassVectorParameterHandler for instance).
-			continue;
-		}
-		if( !m_parameter->parameter<Parameter>( (*pIt)->getName() ) )
+		HandlerMap::iterator nextIt = it; nextIt++; // increment now because removing will invalidate iterator
+		if( std::find( children.begin(), children.end(), it->first ) == children.end() )
```

Comment:
This linear search inside the loop gives us quadratic time complexity again. Is it worth storing children as a flat_set to get faster lookups, or are you happy that this will scale for the maximum numbers of parameters we use?